code/game/gamemodes/wizard/rightandwrong.dm

```diff
@@ -419,7 +420,15 @@
 			new /obj/item/weapon/spellbook/oneuse/ice_barrage(get_turf(src))
 		if("alchemy")
 			new /obj/item/weapon/spellbook/oneuse/alchemy(get_turf(src))
-	new /obj/item/weapon/spellbook/oneuse/absorb(get_turf(src))
+	 
+	var/mob/U = src
+	var/recieve_absorb = 1
```

Comment:
```suggestion
	var/recieve_absorb = !(locate(/spell/targeted/absorb) in U.spell_list)
```
This should work. You can remove the loop entirely and clean up the rest.
Also, it's spelled `receive`.

Reply:
You still need to clean up the rest